ehrapy/api/io/read.py

```diff
@@ -162,15 +158,64 @@ def _read(
                 raw_anndata, columns_obs_only = DataReader.read_csv(
                     filename, delimiter, index_column, columns_obs_only, cache  # type: ignore
                 )
+                # cache results if wanted
+                if cache:
+                    if not path_cache.parent.is_dir():
+                        path_cache.parent.mkdir(parents=True)
+                    return DataReader._write_cache(raw_anndata, path_cache, columns_obs_only)  # type: ignore
+
+            # read from pdf
+            elif extension == "pdf":
+                raw_anndata, columns_obs_only = DataReader.read_pdf(
+                    filename, index_column, columns_obs_only, cache, **kwargs  # type: ignore
+                )
+                # set cache path, since its a single input file which will be stored in (eventually) multiple cache files
+                path_cache = settings.cachedir / filename.stem  # type: Path
+                if cache:
+                    if not path_cache.parent.is_dir():
+                        path_cache.parent.mkdir(parents=True)
+                    path_cache.mkdir()
+                    return DataReader._write_cache_dir(raw_anndata, path_cache, columns_obs_only, index_column)  # type: ignore
+
             else:
                 raise NotImplementedError(f"There is currently no parser implemented for {extension} files!")
-            # cache results if wanted
-            if cache:
-                if not path_cache.parent.is_dir():
-                    path_cache.parent.mkdir(parents=True)
-                return DataReader._write_cache(raw_anndata, path_cache, columns_obs_only)  # type: ignore
             return raw_anndata
 
+    @staticmethod
+    def _read_from_directory(
+        filename: Path,
+        extension: Optional[str] = None,
+        delimiter: Optional[str] = None,
+        index_column: Optional[Union[Dict[str, Union[str, int]], Union[str, int]]] = None,
+        columns_obs_only: Optional[Union[Dict[str, List[str]], List[str]]] = None,
+        return_mudata: bool = False,
+        cache: bool = False,
+        path_cache_dir: Optional[Path] = None,
+    ) -> Dict[str, AnnData]:
+        """Parse AnnData objects from a directory containing the data files"""
+
+        if not extension:
+            raise ExtensionMissingError(
+                "Reading from directory, but no extension has been provided!. Please "
+                "provide an extension for ehrapy to determine, which file format to read!"
```

Comment:
Can we include the list of supported extensions here? 

->

provide one of XXX extensions to blabal